taesungh/pheatures
# Switch to using Vite instead of Create React App

Resolves #1. Removes things related to [react-scripts](https://www.npmjs.com/package/react-scripts) and replaces them with [Vite](https://vite.dev/).
Still need to fix unit testing and enable stricter ESLint rules.

#### Changes
- Begin migration from Create React App to Vite
  - Replace Create React App with Vite based on react-swc-ts template
  - Change index files for reexports to `.ts`
- Use `@/` path alias for imports from `src`
  - Add `baseUrl` & `paths` to `tsconfig.app.json` and update Vite config
- Update Vite config for [importing data files](https://vite.dev/guide/assets.html#explicit-url-imports)
  - Add `.tsv` & `.inv` files to [`assetsInclude`](https://vite.dev/config/shared-options.html#assetsinclude) to import them as paths
- Resolve build and lint issues with TypeScript
  - Will eventually use stricter rules as recommended by ESLint and Vite
- Configure deployment on GitHub Pages
  - Set [`base`](https://vite.dev/guide/build.html#public-base-path) in Vite config to replace `homepage` in `package.json`
  - Recreate deployment workflow based on [latest actions/starter-workflows](https://github.com/actions/starter-workflows/blob/1394e47812550adc70e4af1b714a8000ceee6c00/pages/static.yml)

#### Testing
1. Build the project: `npm run build`
2. Run the preview server: `npm run preview`
3. Confirm the app works as expected
